sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala

```diff
@@ -223,9 +223,9 @@ trait ExpressionEvalHelper extends GeneratorDrivenPropertyChecks with PlanTestBa
           }
         } else {
           val lit = InternalRow(expected, expected)
-          val expectedRow =
-            UnsafeProjection.create(Array(expression.dataType, expression.dataType)).apply(lit)
-          if (unsafeRow != expectedRow) {
+          val dtAsNullable = expression.dataType.asNullable
+          val expectedRow = UnsafeProjection.create(Array(dtAsNullable, dtAsNullable)).apply(lit)
+          if (unsafeRow == expectedRow) {
```

Comment:
Why did you remove the existing test `unsafeRow != expectedRow`?

Reply:
Thank you, it was used for debugging.

Reply:
ah, ok.